### ISSUE TITLE
lsass: GSSAPI Error: The referenced context has expired (Unknown error)

### ISSUE THREAD

**Issue by airinaga** [BeyondTrust/pbis-open#26]
Starting to see these messages occur more frequently in /var/log/messages on all AD authenticated VM's.

Curious if anyone has received these also.

Only relevant information I found was related to VMware vRealize Automation which we do not use.

Thanks,
Adam

**Comment by maxadamo** [comment]
@airinaga does this message cause a delay in the login? 
this happens to me when I login the first time, for instance, in the morning.

**Comment by darzo27** [comment]
**Apologies for placing this issue in completely the incorrect place, I've got an _urgent_ issue to address**

Is anyone aware of a way of creating **Asset based** _Smart Rules_ or _Smart Groups_ via the API ?

Thank you for your time.

**Comment by rbest-bt** [comment]
@darzo27 - please create a new issue since this is unrelated

**Comment by SirIntellegence** [comment]
This keeps spamming my logs as well. I have OSSEC setup and it is annoying to keep receiving emails about something I cannot do anything about.

**Comment by rbest-bt** [comment]
The error is basically saying your kerberos ticket has expired

If you've logged in and didn't enter your password for 8 hours, then your Kerberos session ticket will expire, and may not be renewed. There may be issues with user load or concurrency, which could prevent the ticket from being refreshed. The only reason you need to renew the user's Kerberos ticket is if they're using it for SSO, to another SSH client, to an SMB client (via Nautilis from a workstation desktop), or for NFSv4 mounts, AND if that access is happening 8 hours after the user initially logged in (the default Kerberos expiration time).

If you don't need SSO, you can turn off the following configuration setting (enabled by default), which may improve performance:

>rbest@123456789:~$ /opt/pbis/bin/config --detail RefreshUserCredentials
Name: RefreshUserCredentials
Description: Whether to refresh user credentials agaist AD domain controller
Type: boolean
Current Value: true
Accepted Values: true, false
Current Value is determined by local policy.

**Comment by VigneshVickyS** [comment]
But these message keep added in the system log for every 10mins.

lsass: GSSAPI Error: The referenced context has expired (Unknown error)

There 2 side problems here:
1. Why ticket is acquired for only 8hours?
2. Why reconnect happens only after error, not in advance?

It makes sense in case of kerberos encryption to acquire ticket for longer period (say 24h) and reconnect in advance before ticket expiry, avoiding error message and processing delay.
